pkg/dns/aws/dns.go

```diff
@@ -242,83 +241,102 @@ func (m *Provider) change(record *iov1.DNSRecord, zone configv1.DNSZone, action
 		return fmt.Errorf("unsupported record type %s", record.Spec.RecordType)
 	}
 	// TODO: handle >0 targets
-	domain, target := record.Spec.DNSName, record.Spec.Targets[0]
+	domain, targets := record.Spec.DNSName, record.Spec.Targets
+	updatedRecords := sets.NewString(strings.Join(m.updatedRecords.List(), ","))
 	if len(domain) == 0 {
 		return fmt.Errorf("domain is required")
 	}
-	if len(target) == 0 {
+	if len(targets) == 0 {
 		return fmt.Errorf("target is required")
 	}
 
+	var key string
```

Comment:
Delete this declaration and use `key :=` below to declare `key` inside the loop.

Reply:
done